frontend/js/src/user/components/AddAlbumListens.tsx

```diff
@@ -22,24 +38,49 @@ type MBReleaseWithMetadata = MusicBrainzRelease &
   WithArtistCredits &
   WithReleaseGroup;
 
+/* Allows sorting a selection of tracks based on the original order of that track in the multiple media
+  We cannot just order by the position attribute because it is repeated for each medium (each medium will have a track 1).
+  Courtesy of Shl for this elegant solution: https://stackoverflow.com/a/62298591/4904467
+*/
+function sortByArray<T, U>({
+  source,
+  by,
+  sourceTransformer = identity,
+}: {
+  source: T[];
+  by: U[];
+  sourceTransformer?: (item: T) => U;
+}) {
+  const indexesByElements = new Map(by.map((item, idx) => [item, idx]));
+  const orderedResult = sortBy(source, (p) =>
+    indexesByElements.get(sourceTransformer(p))
+  );
+  return orderedResult;
+}
+
 type TrackRowProps = {
   track: MBTrackWithAC;
   isChecked: boolean;
-  onClickCheckbox: (track: MBTrackWithAC, checked: boolean) => void;
+  onClickCheckbox: (
+    track: MBTrackWithAC,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => void;
 };
 
 function TrackRow({ track, isChecked, onClickCheckbox }: TrackRowProps) {
   return (
     <div className="add-album-track">
       <input
         type="checkbox"
-        onChange={(e) => {
-          onClickCheckbox(track, e.target.checked);
+        onChange={(event) => {
+          onClickCheckbox(track, event);
         }}
         checked={isChecked}
       />
       <strong className="small track-number">
-        {padStart(track.position.toString(), 2, "0")}
+        {isFinite(Number(track.number))
+          ? padStart(track.position.toString(), 2, "0")
+          : track.number}
       </strong>
       <span>{track.title}</span>
       <span className={`duration ${!track.length ? "default-duration" : ""}`}>
```

Comment:
The recording length fallback is still missing in the display component.

Reply:
Do you have a release at hand where a track doesn't have a length but its associated recording does?

Reply:
[This cassette release](https://musicbrainz.org/release/f44999ba-2b55-44f2-b45e-6d956c4116c0) does not have track lengths, for example.

Reply:
Thanks, that allowed me to find another couple of things to fix, looking good now!
![image](https://github.com/metabrainz/listenbrainz-server/assets/6179856/598e5b9a-b87c-4e2c-8d66-e8e78e44569b)